de-lib/src/main/java/org/iplantc/de/teams/client/presenter/EditTeamPresenterImpl.java

```diff
@@ -0,0 +1,302 @@
+    }
+
+    void addMembersToTeam(Group group, IsHideable hideable) {
+        progressDlg.updateProgress();
+        Subject self = createSelfSubject();
+        List<Privilege> privileges = view.getMemberPrivileges();
+        List<Subject> membersToAdd = getSubjectsFromPrivileges(privileges);
+        membersToAdd.add(self);
+        serviceFacade.addMembersToTeam(group, membersToAdd, new AsyncCallback<List<UpdateMemberResult>>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                ErrorHandler.post(throwable);
+                mode = EditTeamView.MODE.EDIT;
+                view.unmask();
+                progressDlg.hide();
+            }
+
+            @Override
+            public void onSuccess(List<UpdateMemberResult> updateMemberResults) {
+                hideable.hide();
+                progressDlg.updateProgress();
+                view.unmask();
+                ensureHandlers().fireEvent(new TeamSaved(group));
+            }
+        });
+    }
+
+    void addPrivilegesToTeam(Group group, List<Privilege> privileges, IsHideable hideable) {
+        progressDlg.updateProgress();
+
+        List<UpdatePrivilegeRequest> updateList = convertPrivilegesToUpdateRequest(privileges);
+
+        UpdatePrivilegeRequestList allUpdates = factory.getUpdatePrivilegeRequestList().as();
+        allUpdates.setRequests(updateList);
+
+        serviceFacade.updateTeamPrivileges(group, allUpdates, new AsyncCallback<List<Privilege>>() {
+            @Override
+            public void onFailure(Throwable throwable) {
+                ErrorHandler.post(throwable);
+                mode = EditTeamView.MODE.EDIT;
+                view.unmask();
+                progressDlg.hide();
+            }
+
+            @Override
+            public void onSuccess(List<Privilege> privileges) {
+                addMembersToTeam(group, hideable);
+            }
+        });
+    }
+
+    Subject createSelfSubject() {
+        Subject subject = factory.getSubject().as();
+        subject.setId(userInfo.getUsername());
+        return subject;
+    }
+
+    @Override
+    public void onUserSearchResultSelected(UserSearchResultSelected userSearchResultSelected) {
+        Subject subject = userSearchResultSelected.getSubject();
+        String tag = userSearchResultSelected.getTag();
+        Privilege privilege = factory.getPrivilege().as();
+        privilege.setSubject(subject);
+        privilege.setPrivilegeType(PrivilegeType.read);
+
+        if (SEARCH_MEMBERS_TAG.equals(tag)) {
+            view.addMembers(Lists.newArrayList(privilege));
+        } else {
+            view.addNonMembers(Lists.newArrayList(privilege));
+        }
+
+    }
+
+    @Override
+    public void onRemoveMemberPrivilegeSelected(RemoveMemberPrivilegeSelected event) {
+        Privilege privilege = event.getPrivilege();
+        if (EditTeamView.MODE.CREATE == mode) {
+            view.removeMemberPrivilege(privilege);
```

Comment:
Ahh, OK, that makes sense now 👍 
Maybe future-us could use a short 1-liner comment here (and below) mentioning something like `remove directly from store. No service calls required in CREATE mode`?